### PR TITLE
Fix permissions for macos 15

### PR DESCRIPTION
**Pull request by DominicVonk**
Thank you for opening a PR! Please make sure that:

* The title and description of the PR convey what the change is about, by mentioning the ticket it addresses for instance.
* The commits messages [follow the convention](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#summary), so your PR can be merged.

### CODE REVIEW

**Comment by DominicVonk** [comment]
Fixes issue #3477

File: src/logic/SystemPermissions.swift [hunk 1]
```diff
@@ -14,25 +14,11 @@ class SystemPermissions {
 
     static func screenRecordingIsGranted() -> Bool {
         if #available(OSX 10.15, *) {
-            return screenRecordingIsGranted_()
+            return CGPreflightScreenCaptureAccess()
         }
         return true
     }
 
-    // workaround: public API CGPreflightScreenCaptureAccess and private API SLSRequestScreenCaptureAccess exist, but
```
Comment (lwouis) [inline src/logic/SystemPermissions.swift:22]: Have you seen this? We were not using this API because it wasn't the best in the past.

Comment (DominicVonk) [reply src/logic/SystemPermissions.swift:22]: Yes, I read it, I will make it that it uses it for OSX 15+

Comment (lwouis) [reply src/logic/SystemPermissions.swift:22]: AltTab checks permissions continuously. The user could remove permissions at any point. We check for that on a repeating timer.

As the comment says, `CGPreflightScreenCaptureAccess` was returning a value which was not updated. It was a snapshot of the permissions at the time the app was launched, but didn't update after that initial snapshot.

Has this behavior changed in macOS 15? If it hasn't, then we should change other parts of the app doing the repeating timer checks. Either find another way to check there, or remove the check. But if we remove the check, we need another way to notice when the `screenshot()` and the AX calls start failing.

Comment (DominicVonk) [reply src/logic/SystemPermissions.swift:22]: I changed it to the canRecordScreen, this is updated during runtime

**Comment by DominicVonk** [comment]
Now I only altered the `private static func screenRecordingIsGranted_()`

I think this is a stable solution

**Comment by lwouis** [comment]
@DominicVonk Using `CGWindowListCopyWindowInfo` is what we used to do years ago.

Here's an [interesting thread to read](https://github.com/lwouis/alt-tab-macos/issues/704#issuecomment-762935477). The whole ticket is about issues with falsely detecting that the permission is gone, when it is in fact not gone.

The current implementation is solid. It has passed the test of time. It's now being challenged by changes in the latest macOS.

Here are some thoughts to go forwards:

* We should keep the existing code for macOS < 15. It's tested and solid.
* For macOS >= 15, I think we should look for answers in [the new API](https://github.com/lwouis/alt-tab-macos/issues/122#issuecomment-1059649102) (this thread also contains a lot worth reading).

**Comment by lwouis** [comment]
I think we can close this PR? In the current state, it doesn't improve things. We still need a break through for better permissions check

**Pull request closed without merging**